applicationinsights-react-js/src/useTrackEvent.ts

```diff
@@ -24,5 +25,11 @@ export default function useCustomEvent<T>(
     reactPlugin.trackEvent({ name: eventName }, data);
   }, [reactPlugin, data, eventName]);
 
+  useEffect(() => {
```

Comment:
Based on the issue, would this not be better as a specific listener for when the component is being "unmounted" as won't this always cause the "firstRun" to get reset directly after it's mounted thereby bypassing the if check on line 21 with a side effect of causing multiple `trackEvent` calls when the `useEffect` is called again (I don't know react well enough to understand if this is even (generally) possible))

Reply:
This is exactly the "unmounted" listener you are describing. That's the reason why it's in a different useEffect hook. If you take a closer look it's not calling the reset inside directly, but it's returning a function that does the work. This function is the "cleanup" function, which will only be called when the component unmount, since it does not have any dependencies. You can read more about it here if you want: https://react.dev/reference/react/useEffect .